doc/syntax.md

```diff
@@ -2,42 +2,323 @@
 
 NESTML has a concise syntax that avoids clutter in the form of
 semicolons, curly braces or tags as known from other programming or
-description languages. Instead it concentrates on the domain concepts
+markup languages. Instead it concentrates on the domain concepts
 needed to efficiently write down neuron models and their equations.
 
 NESTML files are expected to have the filename extension
-`.nestml`. There is no connection between model and file name and each
-file may contain one or more models. Models that shall be compiled
+`.nestml`. There is no connection between model and file name. Each
+file may contain one or more neuron models. Models that shall be compiled
 into one extension module for NEST have to reside in the same
-directory.
+directory. The name of the module corresponds to the name of the folder.
+
+In order to give the users of NESTML complete in implementing neuron's dynamics
+freedom, we provide a full programming language. This programming language
+is used in the `update`-block and `function`-block. It provides the following features:
+
+### Data types and physical units
+Data types define types of variables, function's parameters and
+return values of functions. NESTML provides the following primitive types
+and physical data types:
+#### Primitive data types
+* `real`: corresponds to the `double` data type in C++. Its literals are: 42.0,
+-0.42, .44
+* `integer`: corresponds to the `int` data type in C++. Its literals are: 42, -7
```

Comment:
If we introduce an additional type, we should call it something like integer32 to indicate its size. But right now, I don't think we need it. Configurability might be needed if we start to support other backends than NEST. Right now, I think things can stay as they are.